### PR TITLE
City boxes, "all tiles" page, show company blocker on tile even before it is bought in the private auction

### PR DESCRIPTION
**Pull request by michaeljb**
* Cities with 2 or 3 slots have boxes now (per https://github.com/tobymao/18xx/issues/2#issuecomment-614425795)
  * the 3-slot solution is lazily scaling down the full-size hexagon
  * cities with 4+ slots still don't have a box, but 1889 doesn't use those
* add new route/page at `/all_tiles`, to show all defined tiles
  * also shows all custom starting hexes defined for 1889 (including corporations and company blockers)
* company blocker is now rendered on map even before it is bought during the private auction

### CODE REVIEW

File: api.rb [hunk 1]
```diff
@@ -127,6 +127,10 @@ def on_close(room, q)
       r.halt 404 unless (game = Game[id])
       render(game_data: game.to_h(include_actions: true))
     end
+
+    r.on 'all_tiles' do
+      render(app_route: 'all_tiles')
```
Comment (tobymao) [inline api.rb:132]: you can just put this in the main  on line 94

File: assets/js/view/part/city.rb [hunk 2]
```diff
@@ -61,13 +61,53 @@ def render_part
...
         end
 
-        h(:g, { attrs: { class: 'city' } }, slots)
+        children = []
+
+        children << render_box(slots.size) if slots.size > 1
+
+        children += slots
+
+        h(:g, { attrs: { class: 'city' } }, children)
```
Comment (tobymao) [inline assets/js/view/part/city.rb:76]: could do h('g.city')

File: assets/js/view/part/city.rb [hunk 3]
```diff
@@ -61,13 +61,53 @@ def render_part
...
+              :polygon,
+              attrs: {
+                points: '100,0 50,-87 -50,-87 -100,-0 -50,87 50,87',
+                fill: 'white',
+                transform: 'scale(0.458)',
+              }
+            )
+          end
+
+        h(:g, { attrs: { class: 'city-box' } }, [
```
Comment (tobymao) [inline assets/js/view/part/city.rb:108]: h('g.city_box') let's use all underscores for classes and ids

File: assets/js/view/part/city.rb [hunk 4]
```diff
@@ -61,13 +61,53 @@ def render_part
...
+      #   scaling the full-size hexagon
+      # - implement for 4, 5, and 6 slot cities
+      def render_box(slots)
+        box =
+          case slots
+          when 2
+            h(
+              :rect,
+              attrs: {
+                width: 2 * SLOT_RADIUS,
```
Comment (tobymao) [inline assets/js/view/part/city.rb:90]: these polygon attrs can be constants

File: assets/js/view/part/city.rb (hunk 2 above)
Comment (michaeljb) [reply assets/js/view/part/city.rb]: 9ad0115

File: assets/js/view/part/city.rb (hunk 3 above)
Comment (michaeljb) [reply assets/js/view/part/city.rb]: 9ad0115

File: assets/js/view/part/city.rb (hunk 4 above)
Comment (michaeljb) [reply assets/js/view/part/city.rb]: 5c6d36d

File: api.rb (hunk 1 above)
Comment (michaeljb) [reply api.rb]: 3c5cb1b

**Comment by michaeljb** [comment]
Huh, something in 5c6d36d0ba35125b6d77f81ccc66f1119604559e went haywire...
Screenshot from 2020-04-20 20-17-52

**Comment by michaeljb** [comment]
Fixed in b0593e961138a4dfcf57372fff49aedcca4653e0

**Pull request merged**